### PR TITLE
Added invoices and adjusted code to new db logic

### PR DESCRIPTION
**Pull request by Ghostyshka**

### CODE REVIEW

File: VetClinicManagament/Core/Domain/Entities/Invoice.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using Domain.Enum;
+
+namespace Domain.Entities;
+
+public class Invoice
+{
+    public int InvoiceId { get; set; }
+
+    public int VisitId { get; set; }
+    public Visit Visit { get; set; } // Single Visit Reference
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Entities/Invoice.cs:10]: ````// Single Visit Reference```` 
no need for such comments

File: VetClinicManagament/Core/Domain/Entities/Invoice.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
+    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
+    public InvoiceStatus InvoiceStatus { get; set; }
+
+    public ICollection<InvoiceItem> InvoiceItems { get; set; } // Collection of Invoice Items
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Entities/Invoice.cs:16]: no need for such comments

File: VetClinicManagament/Core/Domain/Entities/Medicals.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public int MedicalId { get; set; }
+    public string MedicalName { get; set; }
+    public int MedicalTypeId { get; set; }
+    public decimal MedicalPrice { get; set; }
+    public int MedicalQuantity { get; set; }
+    public DateTime ExpirationDate { get; set; }
+
+
+    //public MedicalsType MedicalsType { get; set; }
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Entities/Medicals.cs:13]: please fix or remove commented code

File: VetClinicManagament/Core/Domain/Entities/MedicalsType.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Domain.Entities;
+
+public class MedicalsType
+{
+    public int TypeId { get; set; }
+    public string TypeName { get; set; }
+    public int Dosage { get; set; }
+
+    //public ICollection<Medicals> Medicals{ get; set; }
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Entities/MedicalsType.cs:9]: please fix or remove commented code

File: VetClinicManagament/Core/Domain/Entities/Services.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Domain.Entities;
+
+public class Service
+{
+    public int ServiceId { get; set; }
+    public string ServiceTypeId { get; set; }
+    public int MedicalId { get; set; }
+    public decimal Price { get; set; }
+
+    //public ICollection<Treatment> Treatments { get; set; }
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Entities/Services.cs:10]: please fix or remove commented code

File: VetClinicManagament/Core/Domain/Entities/TreatmentPlanItem.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    public int PlanId { get; set; }
+    public int ServiceId { get; set; }
+    public int MedicalId { get; set; }
+    public string ItemDescription { get; set; }
+    public double Dosage { get; set; }
+    public int Quantity { get; set; }
+
+
+
+    //public int AnimalId { get; set; }
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Entities/TreatmentPlanItem.cs:15]: please fix or remove commented code

File: VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs [hunk 7]
```diff
@@ -1,18 +1,19 @@
...
 
 namespace Domain.Models.Dtos;
 
 public class AnimalDto
 {
     private AnimalGender _gender;
 
-    public int OwnerId { get; set; }
-    public int AnimalId { get; set; }
+    public string EmployeeEmail { get; set; }
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs:10]: why do you need EmployeeEmail here?

File: VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs [hunk 8]
```diff
@@ -1,18 +1,19 @@
...
+    public string UserEmail { get; set; }
     public string AnimalName { get; set; }
-    public string Species { get; set; }
-    public int Age { get; set; }
-    public string Breed { get; set; }
-    public string Color { get; set; }
+    public int SpeciesId { get; set; }
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs:13]: why do you need SpeciesId if it already will be in the Specie property below?

File: VetClinicManagament/Core/Domain/Models/AnimalModel.cs [hunk 9]
```diff
@@ -3,7 +3,7 @@
 
 namespace Domain.Models;
 
-public class AnimalModel
+public class AnimalModel        // need to refactor
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Models/AnimalModel.cs:6]: please make it TODO comment and try to implement "TODO" comments before creating pull requests

File: VetClinicManagament/Core/Domain/Repositories/IPersonRepository.cs [hunk 10]
```diff
@@ -0,0 +1,10 @@
+﻿using Domain.Interfaces;
+
+namespace Domain.Repositories;
+
+public interface IPersonRepository
+{
+    Task AddPersonAsync<T>(T person) where T : class;
+    Task<T?> GetPersonByEmailAsync<T>(string email) where T : class, IPerson;
+    //Task GetPersonByEmailAsync(object email); // Optional, for non-generic access
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Repositories/IPersonRepository.cs:9]: please remove commented code

File: VetClinicManagament/Core/Domain/Repositories/IUserRepository.cs [hunk 11]
```diff
@@ -1,15 +1,18 @@
...
 
 namespace Domain.Repositories;
 
 public interface IUserRepository
 {
     Task<int> AddUserAsync(UserRegistrationDto newUser);
-    Task<UserModel> GetUserByEmailAsync(string email);
+    Task<int> AddEmployeeAsync(EmployeeRegistrationDto newEmployee);
+
+    //Task<UserModel> GetUserByEmailAsync(string email);
```
Comment (denyshfive) [inline VetClinicManagament/Core/Domain/Repositories/IUserRepository.cs:12]: please remove commented code

File: VetClinicManagament/Infrastructure/Persistence/Repositories/PersonRepository.cs [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            _dataContext.Set<T>().Add(person);
+            await _dataContext.SaveChangesAsync();
+        }
+        else
+        {
+            throw new InvalidOperationException("Unsupported entity type.");
+        }
+    }
+
+    //public async Task<T?> GetPersonByEmailAsync<T>(string email) where T : class
```
Comment (denyshfive) [inline VetClinicManagament/Infrastructure/Persistence/Repositories/PersonRepository.cs:30]: please remove commented code

File: VetClinicManagament/Infrastructure/Persistence/Configurations/AnimalOwnersConfiguration.cs [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    public void Configure(EntityTypeBuilder<AnimalOwner> builder)
+    {
+        builder.HasKey(ao => ao.Id);
+
+        builder.Property(s => s.UserId)
+            .IsRequired();        
+        
+        builder.Property(s => s.AnimalId)
+            .IsRequired();
+        //check to:do
```
Comment (denyshfive) [inline VetClinicManagament/Infrastructure/Persistence/Configurations/AnimalOwnersConfiguration.cs:18]: please remove

File: VetClinicManagament/Infrastructure/Persistence/Configurations/InvoiceConfiguration.cs [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+        builder.Property(i => i.CreatedAt)
+            .IsRequired();
+
+        builder.Property(i => i.UpdatedAt)
+            .IsRequired();
+
+        builder.Property(i => i.InvoiceStatus)
+            .IsRequired();
+
+        builder.HasMany(i => i.Visits)  // invoice has many visits
```
Comment (denyshfive) [inline VetClinicManagament/Infrastructure/Persistence/Configurations/InvoiceConfiguration.cs:25]: please remove comments

File: VetClinicManagament/Infrastructure/Persistence/Configurations/InvoiceItemConfiguration.cs [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+        builder.HasKey(it => it.ItemId);
+
+        builder.Property(it => it.InvoiceId)
+            .IsRequired();
+
+        builder.Property(it => it.ItemType)
+            .HasMaxLength(128)
+            .IsRequired();
+
+        builder.HasOne(it => it.Invoice)  // each InvoiceItem belongs to one Invoice
```
Comment (denyshfive) [inline VetClinicManagament/Infrastructure/Persistence/Configurations/InvoiceItemConfiguration.cs:20]: please remove comments

File: VetClinicManagament/Infrastructure/Persistence/Configurations/TreatmentPlanConfiguration.cs [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+        builder.Property(tp => tp.StartOfTreatment)
+            .IsRequired();
+
+        builder.Property(tp => tp.EndOfTreatment)
+            .IsRequired();        
+        
+        builder.Property(tp => tp.InClinic)
+            .IsRequired();
+
+        //builder.HasOne(tp => tp.Animal)
```
Comment (denyshfive) [inline VetClinicManagament/Infrastructure/Persistence/Configurations/TreatmentPlanConfiguration.cs:28]: please fix or remove commented code

**Review by denyshfive — CHANGES_REQUESTED** [review]

File: VetClinicManagament/Core/Domain/Models/AnimalModel.cs (hunk 9 above)
Comment (Ghostyshka) [reply VetClinicManagament/Core/Domain/Models/AnimalModel.cs:6]: okay, I understood
thanks for pointing that out

File: VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs (hunk 7 above)
Comment (Ghostyshka) [reply VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs:10]: deleted, thx

File: VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs (hunk 8 above)
Comment (Ghostyshka) [reply VetClinicManagament/Core/Domain/Models/Dtos/AnimalDto.cs:13]: deleted, thx

**Pull request merged**